src/jacobian.jl

```diff
@@ -144,3 +144,14 @@ function jacobian_autodiff(f, x::AbstractArray, nonlinfun, alg)
                                 jac_prototype = jac_prototype, chunksize = chunk_size),
      num_of_chunks)
 end
+
+function jvp(cache::TrustRegionCache{false})
```

Comment:
Put a `!` on there because it's mutating.

Reply:
we are actually not modifying the cache, we are just returning the jvp. Should that still be done?

Reply:
You are modifying the cache in one of them